ambry-replication/src/main/java/com/github/ambry/replication/ReplicationMetrics.java

```diff
@@ -671,6 +699,20 @@ public void updateLagMetricForRemoteReplica(RemoteReplicaInfo remoteReplicaInfo,
     }
   }
 
+  /**
+   * Update catch up point of local replica from the cloud replica.
+   * @param remoteReplicaInfo {@link RemoteReplicaInfo} of the cloud replica.
+   * @param catchUpPoint timestamp upto which local replica has caught with the cloud replica.
+   */
+  public void updateCatchupPointMetricForCloudReplica(RemoteReplicaInfo remoteReplicaInfo, long catchUpPoint) {
+    // update this metric only for cloud peer replica. There will only be one cloud replica peer per partition.
+    if (remoteReplicaInfo.getReplicaId().getReplicaType() == ReplicaType.CLOUD_BACKED
+        && cloudReplicaCatchUpPoint.containsKey(remoteReplicaInfo.getLocalReplicaId().getPartitionId())) {
```

Comment:
Not sure if its possible for there to be a conflict with removeCatchupPointMetricForPartition being called concurrently, but if there is, you could use `cloudReplicaCatchUpPoint.computeIfPresent(remoteReplicaInfo.getLocalReplicaId().getPartitionId(), (k, v) -> catchUpPoint)` instead.

Reply:
As discussed offline, since remove only removes the metric from the registry and not from the map, there shouldn't be any conflict here.